DevSkim-DotNet/Microsoft.DevSkim.CLI/Commands/SuppressionCommand.cs

```diff
@@ -0,0 +1,126 @@
+// Copyright (C) Microsoft. All rights reserved. Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Microsoft.ApplicationInspector.RulesEngine;
+using Microsoft.CodeAnalysis.Sarif;
+using Microsoft.DevSkim.CLI.Options;
+
+namespace Microsoft.DevSkim.CLI.Commands
+{
+    public class SuppressionCommand
+    {
+        private readonly SuppressionCommandOptions _opts;
+
+        public SuppressionCommand(SuppressionCommandOptions options)
+        {
+            _opts = options;
+        }
+
+        public int Run()
+        {
+            var sarifLog = SarifLog.Load(_opts.SarifInput);
+            if (sarifLog.Runs.Count > 0)
+            {
+                var run = sarifLog.Runs[0];
+                var groupedResults = run.Results.GroupBy(x => x.Locations[0].PhysicalLocation.ArtifactLocation.Uri);
+                if (!_opts.ApplyAllSuppression && !_opts.FilesToApplyTo.Any() && !_opts.RulesToApplyFrom.Any())
+                {
+                    Console.WriteLine("Must specify either apply all suppression comments or a combination of file and rules to apply");
+                    return (int)ExitCode.CriticalError;
+                }
+
+                if (_opts.FilesToApplyTo.Any())
+                {
+                    groupedResults =
+                        groupedResults.Where(x => _opts.FilesToApplyTo.Any(y => x.Key.AbsolutePath.Contains(y)));
+                }
+
+                if (_opts.RulesToApplyFrom.Any())
+                {
+                    groupedResults = groupedResults.Where(x =>
+                        _opts.RulesToApplyFrom.Any(y =>
+                            x.Any(z =>
+                                z.RuleId == y)));
+                }
+
+                groupedResults = groupedResults.ToList();
+                foreach (var resultGroup in groupedResults)
+                {
+                    var fileName = resultGroup.Key;
+                    var potentialPath = Path.Combine(_opts.Path, fileName.OriginalString);
+                    // Flatten all the replacements into a single list
+                    var listOfReplacements = resultGroup
+                      .Where(x => x.Locations is { })
+                      .SelectMany(x => x.Locations).Select(x => x.PhysicalLocation).ToList();
+
+                    listOfReplacements
+                    .Sort((a, b) => a.Region.StartLine - b.Region.StartLine);
+
+                    var distinctReplacements  = listOfReplacements
+                    .GroupBy(x => x.Region.StartLine)
+                    .Select(x => x.FirstOrDefault());
+
+                    var ruleIds = string.Join(",", resultGroup.Select(x => x.RuleId).Distinct());
```

Comment:
I may be misunderstanding, but won't this suppress all of the issues detected anywhere in the file on every line where any issue was detected?

Reply:
You are right. The tests surfaced that, I have updated the code to handle multiple issues in the same line.